libeos-payg/tests/plugins/test-provider.c

```diff
@@ -0,0 +1,319 @@
+                                                         gpointer iface_data);
+static void epg_test_provider_provider_iface_init (gpointer g_iface,
+                                                   gpointer iface_data);
+
+static void epg_test_provider_get_property (GObject      *object,
+                                            guint         property_id,
+                                            GValue        *value,
+                                            GParamSpec   *pspec);
+static void epg_test_provider_set_property (GObject      *object,
+                                            guint         property_id,
+                                            const GValue *value,
+                                            GParamSpec   *pspec);
+static void epg_test_provider_constructed  (GObject      *object);
+
+static void epg_test_provider_init_async  (GAsyncInitable      *initable,
+                                           int                  priority,
+                                           GCancellable        *cancellable,
+                                           GAsyncReadyCallback  callback,
+                                           gpointer             user_data);
+static gboolean epg_test_provider_init_finish (GAsyncInitable  *initable,
+                                               GAsyncResult    *result,
+                                               GError         **error);
+
+static gboolean    epg_test_provider_add_code   (EpgProvider  *provider,
+                                                 const gchar  *code_str,
+                                                 guint64       now_secs,
+                                                 GError      **error);
+static gboolean    epg_test_provider_clear_code (EpgProvider   *provider,
+                                                 GError      **error);
+
+static void        epg_test_provider_shutdown_async  (EpgProvider           *provider,
+                                                      GCancellable         *cancellable,
+                                                      GAsyncReadyCallback   callback,
+                                                      gpointer              user_data);
+static gboolean    epg_test_provider_shutdown_finish (EpgProvider           *provider,
+                                                      GAsyncResult         *result,
+                                                      GError              **error);
+
+static guint64     epg_test_provider_get_expiry_time     (EpgProvider *provider);
+static gboolean    epg_test_provider_get_enabled         (EpgProvider *provider);
+static guint64     epg_test_provider_get_rate_limit_end_time (EpgProvider *provider);
+
+typedef struct
+{
+  gboolean enabled;
+} EpgTestProviderPrivate;
+
+typedef enum
+{
+  PROP_EXPIRY_TIME = 1,
+  PROP_ENABLED,
+  PROP_RATE_LIMIT_END_TIME,
+  PROP_CODE_FORMAT,
+} EpgProviderAngazaProperty;
```

Comment:
I suspect this should be `EpgTestProviderProperty`.

Reply:
I think you're right!